src/state-manager/TransactionConsensus.ts

```diff
@@ -1024,6 +1094,565 @@ class TransactionConsenus {
+          }
+          const nodesToSendTo: Set<Shardus.Node> = new Set()
+          for (const data of payload.finalState.stateList) {
+            if (data == null) {
+              continue
+            }
+            if (savedAccountIds.has(data.accountId) === false) {
+              continue
+            }
+            const storageNodes = this.stateManager.transactionQueue.getStorageGroupForAccount(data.accountId)
+            for (const node of storageNodes) {
+              nodesToSendTo.add(node)
+            }
+          }
+          if (nodesToSendTo.size > 0) {
+            Comms.sendGossip(
+              'gossip-final-state',
+              payload.finalState,
+              null,
+              null,
+              Array.from(nodesToSendTo),
+              false,
+              4,
+              queueEntry.acceptedTx.txId
+            )
+            nestedCountersInstance.countEvent(`processing`, `forwarded final data to storage nodes`)
+          }
+        } finally {
+          profilerInstance.scopedProfileSectionEnd('poqo-data-and-receipt')
+        }
+      }
+    )
+
+    Comms.registerInternal(
+      'poqo-send-receipt',
+      (
+        payload: AppliedReceipt2,
+        _respond: unknown,
+        _sender: unknown,
+        _tracker: string,
+        msgSize: number
+      ) => {
+        profilerInstance.scopedProfileSectionStart('poqo-send-receipt', false, msgSize)
+        try{
+          const queueEntry = this.stateManager.transactionQueue.getQueueEntrySafe(payload.txid)
+          if (queueEntry == null) {
+            /* prettier-ignore */ nestedCountersInstance.countEvent('poqo', 'poqo-send-receipt: no queue entry found')
+            return
+          }
+
+          if (queueEntry.poqoReceipt) {
+            // We've already handled this
+            return
+          }
+
+          if (Math.random() < this.debugFailPOQo) {
+            nestedCountersInstance.countEvent('poqo', 'debug fail wont forward receipt')
+            return
+          }
+
+          if (logFlags.verbose) this.mainLogger.debug(`POQo: Received receipt from aggregator for ${queueEntry.logID} starting CT2 for data & receipt`)
+          const executionGroupNodes = new Set(queueEntry.executionGroup.map((node) => node.publicKey))
+          const hasTwoThirdsMajority = this.verifyAppliedReceipt(payload, executionGroupNodes)
+          if (!hasTwoThirdsMajority) {
+            /* prettier-ignore */ if (logFlags.error) this.mainLogger.error(`Receipt does not have the required majority for txid: ${payload.txid}`)
+            nestedCountersInstance.countEvent('poqo', 'poqo-send-receipt: Rejecting receipt because no majority')
+            return
+          }
+          const receivedReceipt = payload as AppliedReceipt2
+          queueEntry.poqoReceipt = receivedReceipt
+          queueEntry.appliedReceipt2 = receivedReceipt
+          queueEntry.recievedAppliedReceipt2 = receivedReceipt
+          queueEntry.hasSentFinalReceipt = true
```

Comment:
same here if gossip failed, true value here will not represent that it was sent

Reply:
same comments as before.   created ARCH-9  to consider not checking gossip/tell requests at the higher level